src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -4,113 +4,124 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.Framework.Configuration.Internal;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.Framework.Configuration
 {
-    public class ConfigurationSection : IConfiguration
+    public class ConfigurationSection : ConfigurationBase, IConfigurationSection
     {
-        private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
+        private string _key;
 
-        public ConfigurationSection(IList<IConfigurationSource> sources)
+        public ConfigurationSection([NotNull] IList<IConfigurationSource> sources)
```

Comment:
OK, then if per the other thread you make the argument for the binding an `IConfiguration` again you shouldn't need to change those tests and you shouldn't need this constructor.